smartsim/_core/entrypoints/file_operations.py

```diff
@@ -0,0 +1,257 @@
+    .. code-block:: bash
+            python -m smartsim._core.entrypoints.file_operations \
+                copy /absolute/file/source/path /absolute/file/dest/path
+
+    /absolute/file/source/path: Path to directory, or path to file to
+        copy to a new location
+    /absolute/file/dest/path: Path to destination directory or path to
+        destination file
+    """
+    if os.path.isdir(parsed_args.source):
+        shutil.copytree(parsed_args.source, parsed_args.dest, dirs_exist_ok=True)
+    else:
+        shutil.copyfile(parsed_args.source, parsed_args.dest)
+
+
+def symlink(parsed_args: argparse.Namespace) -> None:
+    """
+    Create a symbolic link pointing to the exisiting source file
+    named link.
+
+    Sample usage:
+    .. highlight:: bash
+    .. code-block:: bash
+            python -m smartsim._core.entrypoints.file_operations \
+                symlink /absolute/file/source/path /absolute/file/dest/path
+
+    /absolute/file/source/path: the exisiting source path
+    /absolute/file/dest/path: target name where the symlink will be created.
+    """
+    os.symlink(parsed_args.source, parsed_args.dest)
+
+
+def configure(parsed_args: argparse.Namespace) -> None:
+    """Set, search and replace the tagged parameters for the
+    configure operation within tagged files attached to an entity.
+
+    User-formatted files can be attached using the `configure` argument.
+    These files will be modified during ``Application`` generation to replace
+    tagged sections in the user-formatted files with values from the `params`
+    initializer argument used during ``Application`` creation:
+
+    Sample usage:
+    .. highlight:: bash
+    .. code-block:: bash
+            python -m smartsim._core.entrypoints.file_operations \
+                configure /absolute/file/source/pat /absolute/file/dest/path \
+                tag_deliminator param_dict
+
+    /absolute/file/source/path: The tagged files the search and replace operations
+    to be performed upon
+    /absolute/file/dest/path: The destination for configured files to be
+    written to.
+    tag_delimiter: tag for the configure operation to search for, defaults to
+        semi-colon e.g. ";"
+    param_dict: A dict of parameter names and values set for the file
+
+    """
+    tag_delimiter = parsed_args.tag_delimiter
+
+    decoded_dict = base64.b64decode(parsed_args.param_dict)
+    param_dict = pickle.loads(decoded_dict)
+
+    if not param_dict:
+        raise ValueError("param dictionary is empty")
+    if not isinstance(param_dict, dict):
+        raise TypeError("param dict is not a valid dictionary")
+
+    def _get_prev_value(tagged_line: str) -> str:
+        split_tag = tagged_line.split(tag_delimiter)
+        return split_tag[1]
+
+    def make_substitution(tag_name: str, replacement: str) -> Callable[[str], str]:
+        return lambda s: s.replace(
+            f"{tag_delimiter}{tag_name}{tag_delimiter}", str(replacement)
+        )
```

Comment:
and actually, if you go with option (b) here, can you export a type alias of the mapping that you are expecting so that we can type check anything that we plan on serializing to before sending it over the wire?
```py
ParamMappingType: TypeAlias = dict[str, str | int | float] 
```
That way we can hopefully catch any type inconsistencies before run time!

Reply:
no longer applicable. removed function